discovery-config/src/main/scala/akka/discovery/config/ConfigSimpleServiceDiscovery.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.discovery.config
+
+import akka.actor.ExtendedActorSystem
+import akka.discovery.SimpleServiceDiscovery
+import akka.discovery.SimpleServiceDiscovery.{Resolved, ResolvedTarget}
+import akka.event.Logging
+import com.typesafe.config.Config
+
+import scala.collection.JavaConverters._
+import scala.collection.{breakOut, immutable}
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+object ConfigServicesParser {
+  def parse(config: Config): Map[String, Resolved] = {
+    val byService = config.root().entrySet().asScala.map { en =>
+      (en.getKey, config.getConfig(en.getKey))
+    }.toMap
+
+    byService.map {
+      case (serviceName, full) =>
+        val endpoints = full.getConfigList("endpoints").asScala
+        val resolvedTargets: immutable.Seq[ResolvedTarget] = endpoints.map { c =>
+          val host = c.getString("host")
+          val port = if (c.hasPath("port")) Some(c.getInt("port")) else None
+          ResolvedTarget(host, port)
+        }(breakOut)
+        (serviceName, Resolved(serviceName, resolvedTargets))
+    }
+  }
+}
+
+class ConfigSimpleServiceDiscovery(system: ExtendedActorSystem) extends SimpleServiceDiscovery {
+
+  private val log = Logging(system, getClass)
+
+  private val resolvedServices = ConfigServicesParser.parse(
+    system.settings.config.getConfig(system.settings.config.getString("akka.discovery.akka-config.services-path"))
+  )
+
+  log.debug("Config discovery serving: {}", resolvedServices)
+
+  override def lookup(name: String, resolveTimeout: FiniteDuration): Future[SimpleServiceDiscovery.Resolved] = {
+    // TODO or fail or change the Resolved type to an ADT?
```

Comment:
I think this is good as is… especially given the implementation of the aggregate above. You wouldn't want it to print a noisy error every time you look up a service that is not in the config.

If there were an ADT, then I suppose you would still want to handle the case of a successful empty list for completeness sake, so IMHO there's no reason to add another case to handle.